react/features/base/participants/reducer.js

```diff
@@ -64,62 +63,81 @@ function _participant(state, action) {
     case PARTICIPANT_ID_CHANGED:
         if (state.id === action.oldValue) {
             const id = action.newValue;
-
-            return {
+            const newState = {
                 ...state,
-                id,
-                avatar: state.avatar || _getAvatarURL(id, state.email)
+                id
             };
+
+            if (!newState.avatar) {
+                newState.avatar = getAvatarURL(newState);
```

Comment:
I changed your source code but I don't think I changed the behavior here. What I'm thinking now though is that `getAvatarURL` may take the `id` into account which means that the new `avatar` may be different from the old `avatar` because of the `id` change.